Latex Fixes in crypto.py

sudoWin opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

Adds Latex to Function in crypto.py
https://docs.sympy.org/latest/modules/crypto.html#sympy.crypto.crypto.gm_private_key
#### Brief description of what is fixed or changed


#### Other comments

#18859 
#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

sudoWin committed:
Latex Fixes

sylee957 commented:
LaTeX should be denoted by `$$` instead of backticks from the recent docstring guide.

sudoWin committed:
Changed to dollar instead of backtick

sudoWin commented:
Changed to $ signs.

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2899,42 @@ def _random_coprime_stream(n, seed=None):
-        let a_k be a random non-square
-        (non-quadratic residue) modulo p * q
-        such that jacobi_symbol(a, p * q) = 1
+        such that $jacobi \\_symbol(a, pq) = 1$
```
It's better to have a spacing between `pq` even if it renders the same.

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2899,42 @@ def _random_coprime_stream(n, seed=None):
-        let a_k be a random square
+     if $b_k$ = 0:
+        let $a_k$ be a random square
         (quadratic residue) modulo p * q
```
Why not change p * q to latex

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2899,42 @@ def _random_coprime_stream(n, seed=None):
+
+    $\\cdot$ Call their product $N$.
+
+    $\\cdot$ Given a message as an integer $i$, write i in its
```
Why not change the second i to LaTeX form?

sylee957 commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2899,42 @@ def _random_coprime_stream(n, seed=None):
+    $\\cdot$ Given a message as an integer $i$, write i in its
+    bit representation $b_0$ , $\\dotsc$ , $b_n$ .
+
+    $\\cdot$ For each k,
```
Why not change k to LaTeX form?

sudoWin committed:
Minute Fixes

sudoWin commented:
Fixed the above accidental leftovers

smichr committed:
remove unused keyword

This was not used and use would contradict docstring; users wanting to filter text can use the `check_and_join` function.

smichr commented:
I added a couple of other small changes.

sudoWin commented:
Thanks @smichr  
Do you know why wasn't flake8 showing anything for that unused keyword?

smichr commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2900,42 @@ def _random_coprime_stream(n, seed=None):
-    The idea is that, while jacobi_symbol(a, p * q)
-    can be easily computed (and when it is equal to -1 will
-    tell you that a is not a square mod p * q), quadratic
+    The idea is that, while $jacobi \\_ symbol(a, pq)$
```
```suggestion
    The idea is that, while $jacobi \\_ symbol(a, p q)$
```

smichr commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2900,42 @@ def _random_coprime_stream(n, seed=None):
-    tell you that a is not a square mod p * q), quadratic
+    The idea is that, while $jacobi \\_ symbol(a, pq)$
+    can be easily computed (and when it is equal to $-1$ will
+    tell you that $a$ is not a square mod $pq$), quadratic
```
```suggestion
    tell you that $a$ is not a square mod $p q$), quadratic
```

smichr commented on sympy/crypto/crypto.py:
```diff
@@ -2899,37 +2900,42 @@ def _random_coprime_stream(n, seed=None):
 
-    Moreover, approximately half the numbers coprime to p * q have
-    jacobi_symbol equal to 1. And among those, approximately half
+    Moreover, approximately half the numbers coprime to $pq$ have
```
```suggestion
    Moreover, approximately half the numbers coprime to $p q$ have
```

sudoWin committed:
Further Fixes

sudoWin committed:
Merge branch 'crypto-latex' of github.com:sudoWin/sympy into crypto-latex

Fetching commits done by smichr
